This way if the test fails, the build will fail.
Need to send upstream.
Index: libopendbx-1.4.6/test/odbxtest.sh
===================================================================
--- libopendbx-1.4.6.orig/test/odbxtest.sh	2012-05-06 08:10:59.000000000 -0400
+++ libopendbx-1.4.6/test/odbxtest.sh	2012-11-19 12:21:05.468414873 -0500
@@ -7,6 +7,7 @@
 if ! test -f odbxtest.site
 then
 	echo "No odbxtest.site file found"
+	exit 1
 fi
 
 ODBXAPP="./odbxtest ./odbxplustest"
@@ -28,6 +29,7 @@
 		echo "$1 ERRORS:" >> testresult.err
 		cat test.err >> testresult.err
 		diff -b test.out ref/$1.ref >> testresult.err
+		FAILURE="FAILED"
 	else
 		echo "  $1 OK"
 		echo "  $1 OK" >> testresult.log
@@ -154,5 +156,9 @@
 rm -f test.out
 rm -f test.err
 
-
-exit 0
+if [ -n "$FAILURE" ]
+then
+	exit 1
+else
+	exit 0
+fi
